certbot-nginx/certbot_nginx/_internal/configurator.py

```diff
@@ -905,19 +927,24 @@ def _enable_redirect_single(self, domain, vhost):
             logger.info("Redirecting all traffic on port %s to ssl in %s",
                 self.DEFAULT_LISTEN_PORT, vhost.filep)
 
-    def _enable_ocsp_stapling(self, domain, chain_path):
+    def _enable_ocsp_stapling(self, domain: str,
+                              chain_path: Optional[Union[str, List[str]]]) -> None:
         """Include OCSP response in TLS handshake
 
         :param str domain: domain to enable OCSP response for
         :param chain_path: chain file path
         :type chain_path: `str` or `None`
 
         """
+        if not isinstance(chain_path, str) and chain_path is not None:
```

Comment:
Because the signature of the function itself is associated to the way the enhance logic works.

At high level this function, and the other functions related to enhance, are grouped in the dict `_enhance_func`. The effective function to call is accessed by index and then called with the parameters provided by `enhance()`. The signature of `enhance()` itself is controlled by the `Installer` interface, that `NginxConfigurator` implements, which dictates that `options`, the argument 2 passed to enhance functions, is of type `Optional[Union[List[str], str]]`.

So add the end from mypy perspective, all enhance functions must accept `Optional[Union[List[str], str]]`, and the fact that `_enable_ocsp_stapling` raise an error if the option is a `List[str]` is an implementation detail.

Reply:
thanks for tracing that!